### DISCUSSION TITLE
Should the gcd implementations be replaced by Guava implementations?

### DISCUSSION THREAD

**Discussion by axkr** [Hipparchus-Math/hipparchus#312] [Ideas]
According to this blog entry:
- https://medium.com/@m.langer798/stein-vs-stein-on-the-jvm-c911809bfce1

the Guava `gcd()` implementations are faster than the hipparchus implementations.
Can someone confirm this?

- https://github.com/google/guava/blob/fc21ea79c24ae37b977671c425972bf090d7738c/guava/src/com/google/common/math/IntMath.java#L402
- https://github.com/google/guava/blob/fc21ea79c24ae37b977671c425972bf090d7738c/guava/src/com/google/common/math/LongMath.java#L489

See also this pull request:
- https://github.com/apache/commons-numbers/pull/140

**Comment by maisonobe** [comment]
This seems legit.
I don't remember we changing anything in this part, so we most probably have the same performance bottleneck.
